dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/saml/SamlConfigurationService.java

```diff
@@ -0,0 +1,103 @@
+package edu.harvard.iq.dataverse.authorization.providers.saml;
+
+import com.onelogin.saml2.settings.IdPMetadataParser;
+import com.onelogin.saml2.settings.Saml2Settings;
+import com.onelogin.saml2.settings.SettingsBuilder;
+import edu.harvard.iq.dataverse.persistence.user.SamlIdentityProvider;
+import edu.harvard.iq.dataverse.persistence.user.SamlIdentityProviderRepository;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import io.vavr.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.net.URL;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+@Stateless
+public class SamlConfigurationService {
+    private static final Logger logger = LoggerFactory.getLogger(SamlConfigurationService.class);
+
+    private static final String HOST_PLACEHOLDER = "\\{HOST}";
+
+    private SettingsServiceBean settingsService;
+    private SamlIdentityProviderRepository samlIdpRepository;
+
+    private BiFunction<String, String, Map<String, Object>> parser = (String idpMetadataUrl, String idpEntityId) -> {
+        try {
+            return IdPMetadataParser.parseRemoteXML(new URL(idpMetadataUrl), idpEntityId);
+        } catch (Exception e) {
+            logger.warn("Error while fetching or parsing Idp settings", e);
+            throw new IllegalStateException("Cannot fetch Idp settings", e);
+        }
+    };
+
+        // -------------------- CONSTRUCTORS --------------------
+
+    public SamlConfigurationService() { }
+
+    @Inject
+    public SamlConfigurationService(SettingsServiceBean settingsService, SamlIdentityProviderRepository samlIdpRepository) {
+        this.settingsService = settingsService;
+        this.samlIdpRepository = samlIdpRepository;
+    }
+
+    public SamlConfigurationService(SettingsServiceBean settingsService, SamlIdentityProviderRepository samlIdpRepository,
+                                    BiFunction<String, String, Map<String, Object>> parser) {
+        this(settingsService, samlIdpRepository);
+        this.parser = parser;
+    }
+
+    // -------------------- LOGIC --------------------
+
+    public Saml2Settings buildSettings(SamlIdentityProvider provider) {
+        return buildSettings(readSpSettings(), parser.apply(provider.getMetadataUrl(), provider.getEntityId()));
+    }
+
+    public Saml2Settings buildSettings(String idpEntityId) {
```

Comment:
Method will be executed quite often (on every login). I don't know if we can read xml's from external internet source every time

Reply:
I think that this is something we should discuss during daily meeting.

Reply:
[will be done in the future task]

Reply:
Created task: https://github.com/CeON/dataverse/issues/2145